RealmSwift/LinkingObjects.swift

```diff
@@ -422,6 +429,39 @@ extension LinkingObjects : RealmCollection {
     }
 }
 
+private struct LinkingObjectsHandoverMetadata {
+    var propertyName: String
+    var property: RLMProperty?
```

Comment:
Thinking about this more, it's not clear to me that an `enum` is better. I think it would be better if `RLMObjectBase` also used and enum to represent this state, but it uses optional values that mostly mirror these. If we used an enum with a `cached` and `uncached` case, would we check if `property` exists or if `cachedRLMResults` exists when deciding which case to use? (They ought to give the same results, but this is a weird check.) After rewriting it with an enum, I think that the complexity it add is not worth it since it doesn't align with `LinkingObjectsBase`.